man/tpm2_zgen2phase.1.md

```diff
@@ -0,0 +1,83 @@
+% tpm2_zgen2phase(1) tpm2-tools | General Commands Manual
+
+# NAME
+
+**tpm2_zgen2phase**(1) - Command to enable the TPM to combine data from the
+other party with the ephemeral key generated in the first phase of two-phase
+key exchange protocols.
+
+
+# SYNOPSIS
+
+**tpm2_zgen2phase** [*OPTIONS*]
+
+# DESCRIPTION
+
+**tpm2_zgen2phase**(1) - Command to enable the TPM to combine data from the
+other party with the ephemeral key generated in the first phase of two-phase
+key exchange protocols.
+
+# OPTIONS
+
+  * **-c**, **\--key-context**=_FILE_:
+
+    Context object pointing to ECC key.
+    Either a file or a handle number. See section "Context Object Format".
+
+  * **-p**, **\--key-auth**=_AUTH_:
+
+    The authorization value for the ECC key object.
+
+  * **-s**, **\--scheme**=_ALGORITHM_:
+
+    The key exchange scheme. Optional. Valid options are ecdh or sm2.
+
+  * **-t**, **\--counter**=_NATURALNUMBER_:
+
+    The commit count to determine the key index to use.
```

Comment:
Why are the -t option here different from tpm2_ecephermal?

Reply:
In ephemeral, the counter value is incremented. The -t option there is to give the user an option to specify a descriptive file name that he can later use to invoke the specific key index at the time of running say zgen2phase. Note that the counter value increments with commit as well as ephemeral calls. While it is still a key index, the two can be told apart from a descriptive filename.